### ISSUE TITLE
Separate Package for helpers

### ISSUE THREAD

**Issue by nikhilbadgujar** [simpleidserver/SimpleIdServer#221]
I am unable to find any package which contains only Helper functions. I intend to use only SimpleIdServer.Scim.Helpers.SCIMFilterParser, currently for which I have to import whole SimpleIdServer.Scim package. 
Can you create a separate package for that?

**Comment by simpleidserver** [comment]
Hello,

I'm going to create a Nuget Package to export some Helpers method from the SimpleIdServer.Scim package.
The modifications will be included in the release 2.0.4.

Kind Regards,

Thierry H.

**Comment by simpleidserver** [comment]
The SimpleIdServer.SCIM package has been splitted into two packages:
* SimpleIdServer.SCIM.Domains : Contains the Domain classes
* SimpleIdServer.SCIM.Parser : Contains the logic to parse SCIM filter and transform the SCIMExpression into LINQ Expression Tree.

The changes have been made in the branch "release/2.0.4". The new Nuget packages will be available when the release will be published :)

Kind Regards,

SimpleIdServer

**Comment by simpleidserver** [comment]
Hello,

The release "2.0.4" is published. The Nuget package "SimpleIdServer.Scim.Parser"  can be used to convert SCIM filter into LINQ Expression Tree (URL : https://www.nuget.org/packages/SimpleIdServer.Scim.Parser/)

Kind Regards,

SimpleIdServer

**Comment by nikhilbadgujar** [comment]
Hey, thank you for implementing it so quickly, really appreciate that. 
I was testing the new package (i.e. SimpleIdServer.SCIM.Parser) and found that the parser is returning different output when compared with SimpleIdServer.Scim v2.0.3 SCIMFilterParser's output.

**Setup:**
```
var expression = SCIMFilterParser.Parse("urn:xyz:schemas:extension:1.0:xyzUser.subjectId eq abc");
var attributeExpression = (expression as SCIMComparisonExpression).LeftExpression;
```
**Expected (SimpleIdServer.Scim v2.0.3):**
```
attributeExpression.Name = "urn"
attributeExpression.Child.Name = "xyz"
attributeExpression.Child.Child.Name = "schemas"
.
.
.
attributeExpression.GetFullPath() = "urn.xyz.schemas.extension.1.0.xyzUser.subjectId"
```

**Actual (SimpleIdServer.SCIM.Parser ):**
```
attributeExpression.Name = "urn:xyz:schemas:extension:1.0:xyzUser"
attributeExpression.Child.Name = "urn:xyz:schemas:extension:1.0:subjectId"
attributeExpression.Child.Child = null
attributeExpression.GetFullPath() = "urn:xyz:schemas:extension:1.0:xyzUser.urn:xyz:schemas:extension:1.0:subjectId"
```
Parser is inserting "urn:xyz:schemas:extension:1.0:" in child name, which was not expected.

Is this a desired change in v2.0.4 ?
If no then can you please take a look into this. Thanks in advance.

**Comment by simpleidserver** [comment]
Hello,

Your SCIM Expression `urn:xyz:schemas:extension:1.0:xyzUser.subjectId` must be replaced by `urn:xyz:schemas:extension:1.0:xyzUser:subjectId`. 

Because there is a dot before `subjectId`, the parser thinks the Expression is made of two AttributeExpression.

Kind Regards,

Thierry H.

**Comment by nikhilbadgujar** [comment]
Hi,  
As per https://datatracker.ietf.org/doc/html/rfc7644#section-3.10,
> Complex attributes' sub-attributes are referenced via nested dot
   ('.') notation, i.e., {urn}:{Attribute name}.{Sub-Attribute name}.
   For example, the fully qualified path for a User's givenName is
   "urn:ietf:params:scim:schemas:core:2.0:User:name.givenName".

So maybe the earlier implementation would be more helpful, at least for our case.
What are your thoughts?

**Comment by simpleidserver** [comment]
Hello,

You're right there is a small issue with the SCIMParser Expression.
I fixed the issue in the branch "release/2.0.5".

Kind Regards,

SimpleIdServer
